src/main/scala/ChannelMapper.scala

```diff
@@ -10,10 +11,10 @@ import cde.Parameters
  * channel are contiguous.
  */
 class ChannelAddressMapper(n: Int)(implicit p: Parameters) extends Module {
-  val io = new Bundle {
+  val io = IO(new Bundle {
     val in  = Vec(n, new ClientUncachedTileLinkIO).flip
     val out = Vec(n, new ClientUncachedTileLinkIO)
```

Comment:
Oh duh, my bad. Yeah if you were to wrap in Input or Output it would coerce all of the directions.

Instead of `.flip` you should wrap in `Flipped(...)` though